src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,168 @@
+package core.basesyntax;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final String EXCEPTION_MESSAGE = RegistrationException.class.toString();
+
+    private static final String VALID_LOGIN_FIRST = "firstDefaultLogin";
+    private static final String VALID_LOGIN_SECOND = "secondDefaultLogin";
+    private static final String VALID_LOGIN_THIRD = "thirdDefaultLogin";
+
+    private static final String VALID_PASSWORD_FIRST = "123456";
+    private static final String VALID_PASSWORD_SECOND = "LOOOOOOOOOOOOONG";
+    private static final String VALID_PASSWORD_THIRD = "qwe123ZXC$%^";
+
+    private static final int VALID_AGE_FIRST = 18;
+    private static final int VALID_AGE_SECOND = 19;
+    private static final int VALID_AGE_THIRD = 91;
+
+    private static RegistrationServiceImpl registrationService;
+    private List<User> validUsersList;
+
+    private User validUserBob;
+    private User validUserAlice;
+    private User validUserJohn;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        validUserBob = new User();
+        validUserBob.setLogin(VALID_LOGIN_FIRST);
+        validUserBob.setPassword(VALID_PASSWORD_FIRST);
+        validUserBob.setAge(VALID_AGE_FIRST);
+
+        validUserAlice = new User();
+        validUserAlice.setLogin(VALID_LOGIN_SECOND);
+        validUserAlice.setPassword(VALID_PASSWORD_SECOND);
+        validUserAlice.setAge(VALID_AGE_SECOND);
+
+        validUserJohn = new User();
+        validUserJohn.setLogin(VALID_LOGIN_THIRD);
+        validUserJohn.setPassword(VALID_PASSWORD_THIRD);
+        validUserJohn.setAge(VALID_AGE_THIRD);
+
+        validUsersList = new ArrayList<>();
+        validUsersList.add(validUserBob);
+        validUsersList.add(validUserAlice);
+        validUsersList.add(validUserJohn);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void registerValidUsers_Ok() {
```

Comment:
```suggestion
    void register_ValidUsers_Ok() {
```
please follow the pattern when choosing names to your test methods
fix other methods accordingly